proposal/color-4-new-spaces.md

```diff
@@ -786,6 +845,18 @@ To serialize a non-legacy color `color`:
 
 * Let `space-name` be an unquoted lowercase string of `color`'s space name.
 
+* If `color` has a clamped channel whose value is out-of-bounds:
+
+  * Let `xyz` be the result of [converting] `color` into the `xyz` color space.
+
+  * Emit "color-mix(in ", followed by `space-name`, ", ", the result of
+    serializing `xyz`, and then " 100%, black)".
```

Comment:
I get why we need something like this, but it also feels like a bit of a hack. Does it seem like a reliable solution long term, if we're stuck with it?

Reply:
This is sort of a question for the CSS working group. Maybe the actual questions are twofold:

1. Is there an observable difference in a CSS file between representations of the same color in different spaces? That is, could we get away with just serializing this color as XYZ without the `color-mix()` translation layer?

2. If there _is_ an observable difference, is this translation layer likely to be stable going forward?

Reply:
The other impact to consider here is: authors spend a lot of time in browser dev tools, especially when debugging. When they see this syntax, will it make any sense?

Reply:
Probably not, but if it is observably different and there's no other way to represent out-of-gamut colors in plain CSS I don't see that we have much choice.

Reply:
We could _potentially_ flex on this point, but it would involve compromising one of our other principles. Specifically:

* We could say that out-of-gamut colors just get serialized in a way that clamps in-browser, but this violates our principle that Sass representations of values should serialize to equivalent CSS representations.

* We could say that Sass _never generates_ out-of-gamut colors and just specify `color.to-space()` as clamping or gamut-mapping, but this violates the goal that colors should losslessly round-trip between space conversions which is kind of a foundational assumption for our design of a lot of color utilities.

Both of these would also mean that `color.mix()` wouldn't follow the same semantics as CSS, which in turn means that we'd never be able to add a Sass implementation of CSS's `color-mix()` without violating our core design principle of CSS compatibility.

Given that, I think generating messy output in these edge cases is a lower cost than these alternatives.